sympy/parsing/c/c_parser.py

```diff
@@ -400,16 +391,46 @@ def transform_function_decl(self, node):
 
 
             """
-            token = node.get_tokens()
-            c_ret_type = next(token).spelling
-            if (c_ret_type == 'void'):
-                ret_type = none
-            elif(c_ret_type == 'int'):
-                ret_type = IntBaseType(String('integer'))
-            elif (c_ret_type == 'float'):
-                ret_type = FloatBaseType(String('real'))
+            tokens_spelling = [
+            token.spelling for token in node.get_tokens()]
+
+            curr_ret_type = ''
+            for spelling in tokens_spelling:
+                if spelling == node.spelling:
+                    curr_ret_type = curr_ret_type[:-1]
+                    break
+                curr_ret_type += spelling + ' '
```

Comment:
I just checked, you can use `node.result_type` to get the return type for the function and compare `node.result_type.kind` with self._data_types`. There's no need for a new dict either

Reply:
Woah... that's great indeed! No need of that dict now...